subprojects/docs/src/docs/userguide/authoring-builds/introduction/plugins.adoc

```diff
@@ -0,0 +1,579 @@
+This allows Gradle to:
+
+* Optimize the loading and reuse of plugin classes.
+* Provide editors with detailed information about the potential properties and values in the build script.
+
+However, the DSL requires that plugins be defined statically.
+
+There are some key differences between the `plugins {}` block mechanism and the "traditional" `apply()` method mechanism.
+There are also some constraints and possible limitations.
+
+[[sec:constrained_syntax]]
+==== Constrained Syntax
+
+The `plugins {}` block does not support arbitrary code.
+
+It is constrained to be idempotent (produce the same result every time) and side effect-free (safe for Gradle to execute at any time).
+
+The form is:
+
+====
+[.multi-language-sample]
+=====
+.build.gradle.kts
+[source,kotlin]
+----
+plugins {
+    `«plugin id»`                                             // <1>
+    id(«plugin id»)                                           // <2>
+    id(«plugin id») version «plugin version» [apply «false»]  // <3>
+}
+----
+<1> for core Gradle plugins
+<2> for core Gradle plugins or plugins already available to the build script
+<3> for binary Gradle plugins that need to be resolved
+=====
+
+[.multi-language-sample]
+=====
+.build.gradle
+[source,groovy]
+----
+plugins {
+    id «plugin id»                                            // <1>
+    id «plugin id» version «plugin version» [apply «false»]   // <2>
+}
+----
+<1> for core Gradle plugins or plugins already available to the build script
+<2> for binary Gradle plugins that need to be resolved
+=====
+====
+
+[.multi-language-text.lang-kotlin]
+Where `«plugin id»`, in case `#1` is a static Kotlin extension property named after the core plugin ID, and in cases `#2` and `#3` is a string.
+`«plugin version»` is also a string.
+The `apply` statement with a `boolean` can disable the default behavior of applying the plugin immediately (e.g., you want to apply it only in `subprojects`).
+
+[.multi-language-text.lang-groovy]
+Where `«plugin id»` and `«plugin version»` must be constant, literal strings, and the `apply` statement with a `boolean` can be used to disable the default behavior of applying the plugin immediately (e.g., you want to apply it only in `subprojects`).
+No other statements are allowed; their presence will cause a compilation error.
+
+The `plugins{}` block must also be a top-level statement in the build script.
+It cannot be nested inside another construct (e.g., an if-statement or for-loop).
+
+[[sec:build_scripts_only]]
+==== Only in build scripts and settings file
+
+The `plugins{}` block can only be used in a project's build script `build.gradle(.kts)` and the `settings.gradle(.kts)` file.
+It cannot be used in script plugins or init scripts.
+
+If the restrictions of the `plugins{}` block are prohibitive, the recommended approach is to apply plugins using the <<#sec:applying_plugins_buildscript,buildscript{} block>>.
```

Comment:
💭 I don't think we should say this, we want to have people use the `plugins {}` block.